src/sonic-config-engine/sonic-cfggen

```diff
@@ -46,19 +62,65 @@ def unique_name(l):
+            elif table_name == 'DEVICE_METADATA':
+                if content['localhost'].has_key('bgp_default_status'):
+                    data_bgp_admin['all'] = (content['localhost']['bgp_default_status'] == 'up')
+
+        output_data = {'bgp_admin_state': data_bgp_admin} if data_bgp_admin else {}
+        return output_data
+
+    @staticmethod
+    def output_to_db(output_data):
+        db_data = {}
+        for key, value in output_data.iteritems():
+            if key == 'bgp_admin_state':
+                for neighbor, state in value.iteritems():
+                    if neighbor == 'all':
+                        if not db_data.has_key('DEVICE_METADATA'):
+                            db_data['DEVICE_METADATA'] = {'localhost': {}}
+                        db_data['DEVICE_METADATA']['localhost']['bgp_default_status'] = 'up' if state else 'down'
+                    else:
+                        if not db_data.has_key('BGP_NEIGHBOR'):
+                            db_data['BGP_NEIGHBOR'] = {}
+                        if not db_data['BGP_NEIGHBOR'].has_key(neighbor):
+                            db_data['BGP_NEIGHBOR'][neighbor] = {}
+                        db_data['BGP_NEIGHBOR'][neighbor]['admin_status'] = 'up' if state else 'down'
+        return db_data
+
+
 def main():
     parser=argparse.ArgumentParser(description="Render configuration file from minigraph data and jinja2 template.")
     group = parser.add_mutually_exclusive_group()
-    group.add_argument("-m", "--minigraph", help="minigraph xml file")
+    group.add_argument("-m", "--minigraph", help="minigraph xml file", nargs='?', const='/etc/sonic/minigraph.xml')
     group.add_argument("-M", "--device-description", help="device description xml file")
     parser.add_argument("-p", "--port-config", help="port config file, used with -m")
-    parser.add_argument("-y", "--yaml", help="yaml file that contains addtional variables", action='append', default=[])
+    parser.add_argument("-y", "--yaml", help="yaml file that contains additional variables", action='append', default=[])
+    parser.add_argument("-j", "--json", help="json file that contains additional variables", action='append', default=[])
     parser.add_argument("-a", "--additional-data", help="addition data, in json string")
+    parser.add_argument("-d", "--from-db", help="read config from configdb", action='store_true')
     group = parser.add_mutually_exclusive_group()
     group.add_argument("-t", "--template", help="render the data with the template file")
     group.add_argument("-v", "--var", help="print the value of a variable, support jinja2 expression")
     group.add_argument("--var-json", help="print the value of a variable, in json format")
-    group.add_argument("--print-data", help="print all data", action='store_true')
+    group.add_argument("-D", "--write-to-db", help="write config into configdb", action='store_true')
```

Comment:
Let's keep the current option setting for dev purposes. I'll later add a "config load" and "config save" command to CLI for users.

---
In reply to: [128327833](https://github.com/Azure/sonic-buildimage/pull/808#discussion_r128327833) [](ancestors = 128327833)